inc/namespace.php

```diff
@@ -338,3 +339,32 @@ function set_wp_user_query_count_total( $null, WP_User_Query $query ) {
 
 	return $null;
 }
+
+/**
+ * Remove user tax meta when deleting user from subsite.
+ *
+ * @param int $user_id ID of the user being removed.
+ * @param int $blog_id ID of the blog the user is being removed from. Default 0.
+ * @return void
+ */
+function remove_user_tax_meta( int $user_id, int $blog_id ) : void {
+	$restore_blog = false;
+	if ( $blog_id !== get_current_blog_id() ) {
+		switch_to_blog( $blog_id );
+		$restore_blog = true;
+	}
+
+	$user_roles = wp_get_object_terms( $user_id, ROLES_TAXONOMY );
+	array_map( function( $user_role ) use ( $user_id ) {
+		wp_remove_object_terms( $user_id, $user_role->term_id, ROLES_TAXONOMY, true );
```

Comment:
This should use a foreach, as it's not actually mapping the data to anything-- it doesn't _make sense_. If you just want to iterate / loop over an array, just use `foreach`.

Reply:
@joehoyle thanks, changes made.